### DISCUSSION TITLE
Accept Cryptocurrency Donations

### DISCUSSION THREAD

**Discussion by pluja** [photoprism/photoprism#1697] [General]
I'd love to donate if I could do it using [Bitcoin](https://bitcoin.org/) or [Monero](https://getmonero.org/)!

**Comment by pluja** [comment]
Reminding this issue! I'd like to donate if this was available.

**Comment by lastzero** [comment]
Our Dogecoin wallet is finally ready! :champagne: :dog: 

URI: dogecoin:DRCBHso4xaE2DRGpWuaif7jma1UwTKcAGq
Address: DRCBHso4xaE2DRGpWuaif7jma1UwTKcAGq

![dogecoin-wallet](https://user-images.githubusercontent.com/301686/108261193-90349c00-7163-11eb-97e2-cbcad36fe23b.png)

**Comment by lastzero** [comment]
What's the best way to accept Bitcoin or Monero?

**Comment by bjarnemagnussen** [comment]
For Bitcoin (including utilizing its Lightning Network) definitely [BTCPayServer](https://btcpayserver.org), which is a self-hosted, open-source cryptocurrency payment processor. 

Make sure to activate the Lightning Network though inside BTCPayServer, which allows accepting bitcoin donations without having to pay a transaction fee!

**Comment by pluja** [comment]
As @bjarnemagnussen said, for bitcoin using a BTCPayServer would be great. 

For Monero you just need to set up a wallet and share your recieving address. Monero is a privacy coin so you don't have to worry about anything; unlike bitcoin (which has many privacy concerns) it is untraceable so there's no problem with sharing your wallet address online.

For Desktop you may use the official [GUI Wallet](https://www.getmonero.org/downloads/#gui), if you prefer to setup a wallet on an Android or iOS device I recommend you using [CakeWallet](https://cakewallet.com/) as it is an awesome wallet developed by an awesome group of devs!

If you want to learn more about the benefits of using Monero over Bitcoin, I recommend you [this read](https://www.monero.how/why-monero-vs-bitcoin)

**Comment by lastzero** [comment]
Thank you very much! Just installed the Monero GUI Wallet - here's our address:

474qhuQzC59g4i7cn8mN9VeRGRJHvChmUhL5Fh53hntah63QzB7ERxEb5woVvJkc1KZuGZ3GsGh7d5w9d6SUBHmrUzGeEWT

![monero-account](https://user-images.githubusercontent.com/301686/108591341-9778c700-7368-11eb-8610-e64e82cc2db1.png)

**Comment by pluja** [comment]
Nice! I'll be donating later today :dango:

**Comment by lastzero** [comment]
Setting up Dogecoin was a nightmare compared to that :) Guess the hype is over anyway... getting a server with enough storage for BTCPayServer is a more challenging task... might take a while.

**Comment by bjarnemagnussen** [comment]
Regarding BTCPayServer you probably saw already that there also exist third-party providers that host the instance such as [BTCPay Jungle](https://btcpayjungle.com/Account/Login), and a directory with all providers [here](https://directory.btcpayserver.org/filter/hosts)?

This could be a solution to just try it out to see how it works before committing to self-hosting it on your own server.

**Comment by lastzero** [comment]
Thanks, we'll check it out 👍

**Comment by lastzero** [comment]
We've set up CakeWallet on a mobile phone - seems we can accept Bitcoin already, here's the address: bc1qnj5c8wpsc34qe92uql75y6w6y702sfrxva0k4z

@pluja Seems it uses a centralized backend, no blockchain download needed whatsoever. Would our Bitcoins be in danger if the CakeWallet server gets hacked?

**Comment by lastzero** [comment]
Added crypto addresses to our docs:

- https://github.com/photoprism/photoprism/blob/develop/README.md
- https://github.com/photoprism/photoprism/blob/develop/SPONSORS.md
- https://docs.photoprism.org/funding/

Sadly, GitHub doesn't make the links clickable...

**Comment by pluja** [comment]
@lastzero Bitcoin is a decentralized protocol, so no server will ever hold your keys if you use the right software (CakeWallet is pretty good as they never store your keys). As long as you, and only you own your private keys (12 word seed) your Bitcoins will always be safe.

If you are usign CakeWallet you may want to have all the wallets (Monero and Bitcoin) on the same app as CakeWallet lets you have both. To do so you can open the right side menu on the CakeWallet app and go to "Wallets", there you will be able to manage your wallets and create a new one or import (restore) an already existing wallet. I would recommend you to import there the wallet you created with GUI (you just need to write the 25 word seed) so you can manage all wallets from CakeWallet app. Otherwise you can keep Cake for Bitcoin and the GUIWallet for Monero. Please tell me what you will do so I can make the donation to the right address (if you decide to change it creating a new wallet instead of importing the old one for any reason).

Using a 3rd party node shouldn't be an issue as long as you are using a trusted node. However if you ever use a malicious node it could manipulate your transactions so they never reach the mempool or other bad stuff. Don't worry about using CakeWallet node to broadcast your transactions as it is a trusted one, many users use it (myself included). However if you ever get more interested in Bitcoin you could run your own node, which is [pretty easy](https://getumbrel.com/) and it helps supporting the network. The same applies to Monero.

**Comment by lastzero** [comment]
@pluja Thanks! We'll keep the addresses published. Added our Monero wallet to CakeWallet as recommended. No issue at all after typing in the seed. Don't particularly trust mobile phones or cloud services, not the app or blockchain in general that worries me.

**Comment by lastzero** [comment]
Might want to support this proposal by @lastmjs: https://github.com/lastmjs/enable-cryptocurrency-in-funding-yml

**Comment by pluja** [comment]
I just made my small Monero donation! I, for sure, will contribute with some more in the near future to support your awesome project.

**Comment by lastzero** [comment]
Our first crypto donation :birthday: Thanks a lot!
Let us know if you need help enabling early-access features.

**Comment by pluja** [comment]
Hi! I've seen this tweet right here: https://nitter.fdn.fr/stchris_/status/1364133972784119809#m

I would say that you shouldn't worry about this and you don't need to justify nothing. Accepting cryptocurrencies is a very good way to allow people from all around the world to donate to your project. Crypto enables cheap, private, secure and borderless near-instantaneous transactions with low fees (0 fees for you as you are recieving). This enables customers all over the world — even those without access to traditional banks, but with access to the Internet — to donate to your project. Allowing borderless donations from everywhere at anytime no matter where the people live, even from places where there may be government censorship or restrictions. This is a good thing.

Also accepting cryptocurrency can help you attract a demographic of people who prefer the simplicity and anonymity of crypto transactions. In addition, some people (just like me) feel safer paying with crypto as compared to PayPal or credit cards.

I think it's important to always be open to new technology. If a someone wants to donate you in crypto, why should you turn them down? It's quite easy to have a wallet open for anyone who wants to donate.

It opens a free (as in freedom) way to donate to your project and this is pretty good. I don't understand where is the trading part as accepting this kind of donations doesn't imply trading in any way. This is like saying that accepting Fiat donations implies you are trading on stocks.

And last but not least, even in the case you mined bitcoin or monero it would't be nothing bad! I would love to see someone mining as this makes the network more decentralized and secure. You don't need to justify the fact that you are accepting cryptocurrencies as a donation method!

**Comment by lastzero** [comment]
Probably didn't know for how long this has been on our todo. He may be right in that we have no business in (any) currency trading or mining (including gold and copper). Don't like what Elon is doing over at Tesla. Mining on our infrastructure is very slow and expensive as well - plus we need all resources for development.

**Comment by lastzero** [comment]
## ℹ️ Information on Cryptocurrency Donations Has Been Moved to Our Site ##

Please note that we have moved all the information to our website at **[photoprism.app/crypto-donations](https://photoprism.app/crypto-donations)** to make it more official and provide additional details on why we are doing this, what the differences and benefits are, and what the process looks like.
